java/ql/src/experimental/Security/CWE/CWE-200/AndroidFileIntentSource.qll

```diff
@@ -0,0 +1,118 @@
+/** Provides summary models relating to file content inputs of Android. */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.frameworks.android.Android
+
+/** The `startActivityForResult` method of Android `Activity`. */
+class StartActivityForResultMethod extends Method {
+  StartActivityForResultMethod() {
+    this.getDeclaringType().getASupertype*() instanceof AndroidActivity and
+    this.getName() = "startActivityForResult"
+  }
+}
+
+/** Android class instance of `GET_CONTENT` intent. */
+class GetContentIntent extends ClassInstanceExpr {
+  GetContentIntent() {
+    this.getConstructedType().getASupertype*() instanceof TypeIntent and
```

Comment:
This seems dubious since you don't know the arguments to the constructor of an arbitrary class descended from Intent. Do you have particular subtypes in mind where you do know the constructor's signature?

Reply:
Thanks @smowton for reviewing this PR.

Android <code>Intent</code> is seldom subclassed and actually I have never seen such an implementation. So we don't really need the subtype check. The query can be simply changed to:
```this.getConstructedType() instanceof TypeIntent```

I will make the change.

Reply:
Changed